### ISSUE TITLE
Numpy compatibility issue

### ISSUE THREAD

**Issue by GalKepler** [dipy/dipy#3299]
## Description
I'm trying to use ReconstDtiFlow to estimate the diffusion tensor but encounter the following error:

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[4], [line 1](vscode-notebook-cell:?execution_count=4&line=1)
----> [1](vscode-notebook-cell:?execution_count=4&line=1) from dipy.workflows.reconst import ReconstDtiFlow

File ~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:9
      [5](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:5) from warnings import warn
      [7](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:7) import nibabel as nib
----> [9](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:9) from dipy.core.gradients import mask_non_weighted_bvals, gradient_table
     [10](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:10) from dipy.data import default_sphere, get_sphere
     [11](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/workflows/reconst.py:11) from dipy.io.gradients import read_bvals_bvecs

File ~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:10
      [8](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:8) from dipy.core.onetime import auto_attr
      [9](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:9) from dipy.core.geometry import vector_norm, vec2vec_rotmat
---> [10](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:10) from dipy.core.sphere import disperse_charges, HemiSphere
     [12](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:12) from dipy.utils.deprecator import deprecate_with_version
     [15](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/gradients.py:15) WATER_GYROMAGNETIC_RATIO = 267.513e6  # 1/(sT)

File ~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:8
      [6](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:6) from dipy.core.geometry import cart2sphere, sphere2cart, vector_norm
      [7](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:7) from dipy.core.onetime import auto_attr
----> [8](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:8) from dipy.reconst.recspeed import remove_similar_vertices
     [10](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:10) __all__ = ['Sphere', 'HemiSphere', 'faces_from_sphere_vertices',
     [11](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:11)            'unique_edges']
     [14](https://vscode-remote+ssh-002dremote-002b132-002e66-002e42-002e173.vscode-resource.vscode-cdn.net/home/galkepler/Projects/kepost/trials/~/Projects/kepost/.venv/lib/python3.10/site-packages/dipy/core/sphere.py:14) def _all_specified(*args):

File recspeed.pyx:1, in init dipy.reconst.recspeed()

ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

TLDR - the problematic function is:

`from dipy.reconst.recspeed import remove_similar_vertices`.

## Way to reproduce
-  Code example
```python
from dipy.reconst.recspeed import remove_similar_vertices
```
Or
```python
from dipy.workflows.reconst import ReconstDtiFlow
```
- Operating system and version
Linux-6.5.0-44-generic-x86_64-with-glibc2.35
- Python version
Python 3.10.12 (main, Mar 22 2024, 16:50:05) [GCC 11.4.0]
- dipy version
1.9.0
- [ ] dependency version (numpy, scipy, nibabel, h5py, cvxpy, fury)
    * NumPy 2.0.1
    * SciPy 1.14.0
    * Nibabel 5.2.1
    * H5py 3.11.0
    * Cvxpy 1.5.2
    * fury 0.10.0

**Comment by GalKepler** [comment]
This issue is fixable by downgrading Numpy to 1.26.4, but I think it's a good idea to have dipy keep track with newer version of numpy.

**Comment by skoudoro** [comment]
Hi @GalKepler,

Yes, it is on the way.  

master branch is compatible with numpy 2.0, but we need to cut a release. should be soon to fix that

**Comment by GalKepler** [comment]
Great, thank you!
